control-plane/api/mesh/v2beta1/gateway_class_config_types.go

```diff
@@ -64,6 +70,36 @@ type GatewayClassDeploymentConfig struct {
 	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
 }
 
+type NodeSelector map[string]string
+
+func (n *NodeSelector) UnmarshalJSON(in []byte) error {
```

Comment:
honestly we could do this either here when read this value in from the yaml file or when we create the deployment